src/streamlink/plugins/vimeo.py

```diff
@@ -65,7 +81,24 @@ def _get_streams(self):
         if "player.vimeo.com" in self.url:
             data = self.session.http.get(self.url, schema=self._player_schema)
         else:
-            api_url = self.session.http.get(self.url, schema=self._config_url_schema)
+            viewer = self.session.http.get(
+                self.VIEWER_URL,
+                schema=self._viewer_schema,
+            )
+
+            uri = self.session.http.get(
+                self.OEMBED_URL,
+                params={"url": self.url},
+                schema=self._uri_schema,
+            )
+
+            if viewer and uri:
+                api_url = self.session.http.get(
+                    self.API_URL.format(viewer["apiUrl"], uri["uri"]),
+                    headers={"Authorization": "jwt {}".format(viewer["jwt"])},
+                    schema=self._config_url_schema,
+                )
```

Comment:
`viewer` and `url` can never be falsy. If the validation schema doesn't validate, then a `PluginError` will be raised, so it's always a dict instance according to the schemas.

`api_url` also must be defined outside of this if-block, as you're checking whether it's truthy down below.

Just remove the unnecessary if-block.

It'd also be much better to refactor all this into a separate method, so the complexity of the `_get_streams()` method it self doesn't increase further with these API queries.